docs/guide/callbacks.rst

```diff
@@ -0,0 +1,168 @@
+            super(CustomCallback, self).__init__(verbose)
+            # Those variables will be accessible in the callback
+            # (they are defined in the base class)
+            # self.model = None
+            # self.training_env = None
+            # self.n_calls = 0
+            # self.num_timesteps = 0
+
+        def on_training_start(self, locals_: dict, globals_: dict) -> None:
+            pass
+
+        def on_step(self, locals_: dict, globals_: dict) -> bool:
+            """ This method will be called by the model.
+            This is the equivalent to the callback function.
+
+            :param locals_: (dict)
+            :param globals_: (dict)
+            :return: (bool)
+            """
+            return True
+
+        def on_training_end(self, locals_: dict, globals_: dict) -> None:
+            pass
+
+
+Callback Collection
+-------------------
+
+Stable Baselines provides you with a set of common callbacks.
+
+CheckpointCallback
+^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    from stable_baselines import SAC
+    from stable_baselines.common.callbacks import CheckpointCallback
+
+    checkpoint_callback = CheckpointCallback(save_freq=1000, save_path='./logs/')
+
+    model = SAC('MlpPolicy', 'Pendulum-v0')
+    model.learn(2000, callback=checkpoint_callback)
+
+
+EvalCallback
+^^^^^^^^^^^^
+
+For proper evaluation, using a separate test environment.
+
+.. code-block:: python
+
+    import gym
+
+    from stable_baselines import SAC
+    from stable_baselines.common.callbacks import EvalCallback
+
+    # Separate evaluation env
+    eval_env = gym.make('Pendulum-v0')
+    eval_callback = EvalCallback(eval_env, best_model_save_path='./logs/best_model',
+                                 log_path='./logs/results', eval_freq=500)
+
+    model = SAC('MlpPolicy', 'Pendulum-v0')
+    model.learn(5000, callback=eval_callback)
+
+
+CallbackList
+^^^^^^^^^^^^
+
+For chaining callbacks.
+
+.. code-block:: python
+
+    import gym
+
+    from stable_baselines import SAC
+    from stable_baselines.common.callbacks import CallbackList, CheckpointCallback, EvalCallback
+
+    checkpoint_callback = CheckpointCallback(save_freq=1000, save_path='./logs/')
+    # Separate evaluation env
+    eval_env = gym.make('Pendulum-v0')
+    eval_callback = EvalCallback(eval_env, best_model_save_path='./logs/best_model',
+                                 log_path='./logs/results', eval_freq=500)
+    # Create the callback list
+    callback = CallbackList([checkpoint_callback, eval_callback])
+
+    model = SAC('MlpPolicy', 'Pendulum-v0')
+    # Equivalent to:
+    # model.learn(5000, callback=[checkpoint_callback, eval_callback])
+    model.learn(5000, callback=callback)
+
+
+LambdaCallback
+^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    import gym
+
+    from stable_baselines import SAC
+    from stable_baselines.common.callbacks import LambdaCallback
+
+    # Dummy callback
+    callback = LambdaCallback(on_training_start=None,
```

Comment:
Could you give an example of when you'd want to use `LambdaCallback`? It seems hard to do much in a `lambda` function. And the user could also just pass a `lambda` function directly (unless we're deprecating callbacks-as-functions as part of this change).

Reply:
I understood this would make it cleaner to do a callback without creating a new class, e.g. for `on_training_start`. Without `LambdaCallback` you would have to, in current form, inherit the callback class and define `on_training_start`. I agree the `on_step` could be as well done with the functional form right now, but callback-class might end up having e.g. `on_rollout_start` callbacks which can be accessed with this `LambdaCallback`.